app/scripts/views/mixins/avatar-mixin.js

```diff
@@ -62,11 +62,10 @@ define([
 
     // Makes sure the account with uid has an uptodate image cache.
     // This should be called after fetching the current profile image.
-    _updateCachedProfileImage: function (uid, profileImage) {
-      var cachedAccount = this.user.getAccountByUid(uid);
-      if (! cachedAccount.isDefault()) {
-        cachedAccount.setProfileImage(profileImage);
-        this.user.setAccount(cachedAccount);
+    _updateCachedProfileImage: function (account, profileImage) {
+      if (! account.isDefault()) {
+        account.setProfileImage(profileImage);
+        this.user.setAccount(account);
```

Comment:
One question I've always had, how come every time an account is updated, `this.user.setAccount` needs to be called? IIRC, it has to do with persistence. Can an account persist itself?